contracts/modules/legacy/LegacyPooledStaking.sol

```diff
@@ -840,160 +840,4 @@ contract LegacyPooledStaking is IPooledStaking, MasterAwareV2, PricesV1 {
 
     return products;
   }
-
-  function createPoolAndDeposit(
-    StakingPoolMigrationData memory migrationData,
-    ProductInitializationParams[] memory products
-  ) internal returns (uint depositMigrated) {
-
-    (uint stakingPoolId, address stakingPoolAddress) = cover.createStakingPool(
-      migrationData.isPrivatePool,
-      migrationData.initialPoolFee,
-      migrationData.maxPoolFee,
-      products,
-      migrationData.ipfsDescriptionHash
-    );
-
-    token().approve(address(tokenController()), migrationData.deposit);
-
-    uint totalStakeRatio = 0;
-    uint tokenId = 0; // 0 means a new NFT will be created that will then be reused for each tranche
-    uint firstTrancheId = block.timestamp / 91 days;
-
-    for (uint i = 0; i < MAX_ACTIVE_TRANCHES; i++) {
-
-      if (migrationData.trancheStakeRatio[i] == 0) {
-        continue;
-      }
-
-      totalStakeRatio += migrationData.trancheStakeRatio[i];
-
-      uint trancheDeposit = totalStakeRatio == 100
-        ? migrationData.deposit - depositMigrated
-        : migrationData.deposit * migrationData.trancheStakeRatio[i] / 100;
-
-      depositMigrated += trancheDeposit;
-
-      tokenId = IStakingPool(stakingPoolAddress).depositTo(
-        trancheDeposit,
-        firstTrancheId + i,
-        tokenId,
-        address(this)
-      );
-    }
-
-    tokenController().assignStakingPoolManager(stakingPoolId, migrationData.managerAddress);
-    stakingNFT.transferFrom(address(this), migrationData.stakerAddress, tokenId);
-
-    return depositMigrated;
-  }
-
-  function migrateToNewV2Pool(address stakerAddress) external noPendingActions {
-    // Only selected stakers are automatically migrated
-    require(
-      stakerAddress == ARMOR_STAKER ||
-      stakerAddress == HUGH ||
-      stakerAddress == NM_FOUNDATION,
-      "You are not authorized to migrate this staker"
-    );
-
-    uint deposit = stakers[stakerAddress].deposit;
-    require(deposit > 0, "Address has no stake to migrate");
-
-    INXMToken nxm = token();
-    uint nxmBalanceBefore = nxm.balanceOf(address(this));
-    uint expectedMigratedAmount = 0;
-
-    ProductInitializationParams[] memory products = getProducts(stakerAddress, deposit);
```

Comment:
I think it's safe to remove the `getProducts` function as well?

Reply:
you're right, removed